### PR TITLE
Timeout added for opener

### LINKED ISSUE #20: How to set the timeout parameter
#### Issue Summary

Maybe I missed something, But I have no idea how to  set timeout of a http request using this library

### PR DESCRIPTION
**Pull request by prabhakar267**
Fixes #20

### CODE REVIEW

**Comment by SendGridDX** [comment]
[CLA assistant check](https://cla.sendgrid.com/sendgrid/python-http-client?pullRequest=42) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.sendgrid.com/sendgrid/python-http-client?pullRequest=42) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.sendgrid.com/check/sendgrid/python-http-client?pullRequest=42) it.</sub>

**Comment by thinkingserious** [comment]
Thank you @prabhakar267!

Please add this parameter to the calling functions so that it can be configurable. 

With Best Regards,

Elmer

**Comment by prabhakar267** [comment]
@thinkingserious I have updated the parameter and all calling functions
Also, I had to update a unit test. :smile:

**Comment by thinkingserious** [comment]
@prabhakar267

We have not been able to merge your Pull Request, but because you are awesome - we wanted to make sure you could still get a SendGrid Hacktoberfest shirt.  

Please go **fill [out our swag form](https://dx.sendgrid.com/swag) before Nov 5th** and we will send the shirt! (We know that you might have tried this before and it didn’t work, sorry about that!)

### **You have till Nov 5th to fill out this form** in order to get the Hacktoberfest shirt!

Thank you for contributing during Hacktoberfest! We hope to see you in the repos soon! Just so you know, we always give away a SendGrid shirt for your first ever non-Hacktoberfest PR that gets merged.

File: python_http_client/client.py [hunk 1]
```diff
@@ -143,18 +143,20 @@ def _build_client(self, name=None):
                       url_path=url_path,
                       append_slash=self.append_slash)
 
-    def _make_request(self, opener, request):
+    def _make_request(self, opener, request, timeout=4):
```
Comment (wynnw) [inline python_http_client/client.py]: This seems incomplete as I don't see a way to get a timeout value down into this call from the public api. It looks to me like the Client.__getattr__ method's nested http_request should take a timeout parameter from the **kwargs and then pass that on to the call to self._make_request. You're hard coding a 4 second timeout here which isn't right - that should be up to the caller. The default timeout shouldn't be set to anything either to stay consistent with how every other api I've used handles timeouts.

**Comment by thinkingserious** [comment]
https://github.com/sendgrid/python-http-client/pull/21#issuecomment-391087056

**Pull request closed without merging**